roles/kube_prometheus_stack/vars/main.yml

```diff
@@ -31,6 +31,24 @@ _kube_prometheus_stack_helm_values:
         openstack-control-plane: enabled
   grafana:
     adminPassword: "{{ kube_prometheus_stack_grafana_admin_password }}"
+    grafana.ini:
+      server:
+        root_url: https://{{ kube_prometheus_stack_grafana_host }}
+      auth.generic_oauth:
+        enabled: true
+        name: Keycloak-OAuth
+        allow_sign_up: true
+        client_id: "{{ atmosphere_grafana_oauth_client_id | default('grafana') }}"
+        client_secret: "{{ atmosphere_grafana_oauth_client_secret }}"
+        scopes: openid email profile offline_access roles
+        email_attribute_path: email
+        login_attribute_path: username
+        name_attribute_path: full_name
+        auth_url: "https://{{ keycloak_host }}/realms/{{ atmosphere_keycloak_realm | default('Atmosphere') }}/protocol/openid-connect/auth"
+        token_url: "https://{{ keycloak_host }}/realms/{{ atmosphere_keycloak_realm | default('Atmosphere') }}/protocol/openid-connect/token"
+        api_url: "https://{{ keycloak_host }}/realms/{{ atmosphere_keycloak_realm | default('Atmosphere') }}/protocol/openid-connect/userinfo"
```

Comment:
If we are actually using the role, we can actually set a fact from creating the client and get that information directly back from it!

Reply:
the `keycloak_realm` module doesn't output openid config url https://docs.ansible.com/ansible/latest/collections/community/general/keycloak_realm_module.html#return-values.
Just we can use the realm id from the `proposed` output.